### PR TITLE
Update data dev quickstart document

### PR DESCRIPTION
**Pull request by SamGreig**
#### 📲 What

Updated data dev quickstart docs to be more informative.

#### 🤔 Why
		
Why it's needed, background context.
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 Evidence
		
Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR.
		 
#### 🕵️ How to test

Notes on how a reviewer can test the changes, e.g. how to run the tests.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ To run the E2E tests, you need to set up the following environment variables:
 - `AZURE_STORAGE_ACCOUNT_NAME`
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md:55]: Env variables should be updated:
AZURE_SUBSCRIPTION_ID
AZURE_RESOURCE_GROUP_NAME
AZURE_DATA_FACTORY_NAME
AZURE_REGION_NAME
AZURE_STORAGE_ACCOUNT_NAME

File: docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Please execute the following steps to start working locally with Stacks Azure Da
 
 1. Make sure you have installed the applications from the "Local development" section listed in
 [Requirements](../requirements_data_azure.md).
-2. Ensure that Poetry is added to your `$PATH`.
+2. Ensure that Poetry is added to your `$PATH`. If this results in errors whilst running the tests, we recommend installing Poetry using the [pip CLI](https://pypi.org/project/pip/).
```
[the remark this answers is not in the document]
Comment (SamGreig) [reply docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md]: Resolved by adding troubleshooting section.

File: docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ To run the E2E tests, you need to set up the following environment variables:
 - `AZURE_STORAGE_ACCOUNT_NAME`
 
 The E2E tests may require additional permissions as we are editing data in ADLS during the E2E tests. If the tests fail
-whilst clearing up directories please check you have the necessary permissions to read, write and execute against ADLS.
+whilst clearing up directories you will need to have [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor) access applied to your Azure Active Directory subscription. You may also be required to configure the [firewall rules](https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security) for the storage account to whitelist your IP address.
```
[the remark this answers is not in the document]
Comment (SamGreig) [reply docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md]: Resolved in latest push.

File: docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md (hunk 1 above)
Comment (SamGreig) [reply docs/workloads/azure/data/getting_started/dev_quickstart_data_azure.md:55]: Resolve in latest push.

**Review by motylp — APPROVED** [review]

**Pull request merged**